### PR TITLE
[Doc] Create STSELib documentation

### PR DESCRIPTION
**Pull request by Grom-**

### CODE REVIEW

File: doc/resources/dox_files/APIs/device_management/stse_device_unlock.dox [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+\n\n \b Use-case \b example 
+\n The following applicative code snippet illustrates how to use this API function in main application.
+\n\n
+
+\code{.c}
+int main (void)
+{
+    stsafe_ReturnCode_t stsafe_ret;
+    PLAT_UI8 password[STSAFE_PASSWORD_LENGTH] = { /* password value set during device personalization*/ };
```
Comment (Vaarai) [inline doc/resources/dox_files/APIs/device_management/stse_device_unlock.dox:33]: Replace STSAFE by STSE

File: doc/resources/dox_files/APIs/device_management/stse_device_unlock.dox [hunk 2]
```diff
@@ -0,0 +1,46 @@
+\b Description 
+Following diagram illustrates the interactions performed between the Host and the target STSAFE-A1xx device during the API execution
```
Comment (Vaarai) [inline doc/resources/dox_files/APIs/device_management/stse_device_unlock.dox:2]: Replace STSAFE-A1xx by STSE

File: doc/resources/dox_files/APIs/device_management/stse_init.dox [hunk 3]
```diff
@@ -0,0 +1,65 @@
+\b Description 
+Following diagram illustrates the interactions performed between the Host and the target STSAFE-A1xx device during the API execution
```
Comment (Vaarai) [inline doc/resources/dox_files/APIs/device_management/stse_init.dox:2]: Replace STSAFE-A1xx by STSE

File: doc/resources/dox_files/Services/stsafe_verify_password.dox [hunk 4]
```diff
@@ -8,7 +8,7 @@ Following diagram illustrates the interactions performed between the Host and th
     participant "HOST" as HOST
     participant "STSAFE-Axxx" as STSAFE
```
Comment (Vaarai) [inline doc/resources/dox_files/Services/stsafe_verify_password.dox:9]: Replace by:
```C
participant "STSE" as STSE
```

File: doc/resources/doxygen-awesome/LICENSE
Comment (Vaarai) [inline doc/resources/doxygen-awesome/LICENSE:1]: doxygen-awesome is a separate module, STSELib require an SBOM

File: services/stsafea/stsafea_aes.h [hunk 5]
```diff
@@ -30,7 +30,7 @@
 #include "services/stsafea/stsafea_symmetric_key_slots.h"
 #include "services/stsafea/stsafea_sessions.h"
 
-/*! \defgroup stsafea_aes AES
+/*! \defgroup stsafea_aes STSAFE AES services
```
Comment (Vaarai) [inline services/stsafea/stsafea_aes.h:33]: STSAFE`-A` AES services

File: services/stsafea/stsafea_data_partition.h [hunk 6]
```diff
@@ -28,7 +28,7 @@
 #include "services/stsafea/stsafea_timings.h"
 #include "services/stsafea/stsafea_put_query.h"
 
-/*! \defgroup stsafe_data_partition Data partition management
+/*! \defgroup stsafe_data_partition STSAFE-A Data partition management
```
Comment (Vaarai) [inline services/stsafea/stsafea_data_partition.h:31]: `stsafe_data_partition` => `stsafea_data_partition`

**Review by Vaarai — CHANGES_REQUESTED** [review]
As a gereral feedback I strongly recommend to replace all `STSAFE*` occurences in `API/` layer.

File: services/stsafea/stsafea_data_partition.h (hunk 6 above)
Comment (Grom-) [reply services/stsafea/stsafea_data_partition.h:31]: Updated

**Review by Vaarai — CHANGES_REQUESTED** [review]
Added few comments and some comments remains from previous review.

STSAFE keyword is still used in API layer where it should be replaced by STSE.

File: doc/resources/dox_files/Services/stsafe_verify_password.dox (hunk 4 above)
Comment (Grom-) [reply doc/resources/dox_files/Services/stsafe_verify_password.dox:9]: We are in the STSAFE-A service layer

File: services/stsafea/stsafea_aes.h (hunk 5 above)
Comment (Grom-) [reply services/stsafea/stsafea_aes.h:33]: Done

File: doc/resources/dox_files/Core/__template.dox
[the remark this answers is not in the document]
Comment (Grom-) [reply doc/resources/dox_files/Core/__template.dox:1]: Done

File: services/stsafea/stsafea_sessions.h [hunk 7]
```diff
@@ -30,7 +30,7 @@
  * \param[in] 	*pSession 			\ref stse_session_t Pointer to session
  * \param[in] 	*pHost_MAC_key 		Pointer to MAC key buffer to be used under the session
  * \param[in] 	*pHost_cypher_key 	Pointer to cypher key buffer to be used under the session
- * \return \ref stsafe_ReturnCode_t : STSAFE_OK on success ; error code otherwise
+ * \return \ref stsafe_ReturnCode_t : STSE_OK on success ; error code otherwise
```
[the remark this answers is not in the document]
Comment (Grom-) [reply services/stsafea/stsafea_sessions.h:33]: Done

File: services/stsafea/stsafea_sessions.h [hunk 8]
```diff
@@ -42,14 +42,14 @@ stse_ReturnCode_t stsafea_open_host_session( stse_Handler_t *pSTSE,
...
  */
 void stsafea_close_host_session(stse_session_t *pSession);
 
 /*!
  * \brief 		This Core function Erase the session context from STSAFE handler
  * \param[in] 	*pSTSE 	Pointer to target STSAFE handler
- * \return \ref stsafe_ReturnCode_t : STSAFE_OK on success ; error code otherwise
+ * \return \ref stsafe_ReturnCode_t : STSE_OK on success ; error code otherwise
```
[the remark this answers is not in the document]
Comment (Grom-) [reply services/stsafea/stsafea_sessions.h:52]: Done

**Review by Vaarai — APPROVED** [review]
Good job !!!

**Comment by Vaarai** [comment]
PR merged, @Grom-  you can delete your branch safely.

**Pull request merged**